popde/density_estimate.py

```diff
@@ -120,6 +120,76 @@ def prepare_data(self):
         else:
             self.kde_data = self.std_data
 
+
+    def evaluate_with_transf(self, points):
+        """
+        Transforms the input points in the same way as the KDE training data,
+        evaluates the KDE on the transformed points and adjusts the KDE
+        values by the Jacobian of the transformations.
+
+        Parameters:
+        -----------
+        points : array-like, shape (n_samples, n_features)
+            The original parameter space points to be transformed and evaluated.
+
+        Returns:
+        --------
+        kde_vals : array-like, shape (n_samples,)
+        The KDE values adjusted by the Jacobian of the transformations.
+        """
+        if self.input_transf is not None:
+            transf_points = transf.transform_data(points, self.input_transf)
+        else:
+            transf_points = points
+        #normalize
+        if self.stdize:
+            std_transf = ['stdize' for dim in self.ndim]
+            std_values_for_points = np.std(transf_points, axis=0)  # record the stds
+            std_points = transf.transform_data(transf_points, std_transf)
+        else:
+            std_values_for_points = None
+            std_points = transf_points
+        #rescale
+        if self.rescale is not None:
+            transf_data = transf.transform_data(std_points, self.rescale)
+        else:
+            transf_data = std_points
+
+        #Evaluate kde on transform points
+        kde_vals = self.evaluate(transf_data)
+
+        # Jacobian of transform
+        for i, option in enumerate(self.input_transf):
+            if option in['log', 'ln']:
+                input_Jacobian = 1.0 / points[:, i]
+            elif option == 'exp':
+                input_Jacobian = np.exp(points[:, i])
+            elif option in ('none', 'None'):
+                #print("no need for Jacobian")
+                input_Jacobian  = points[:, i]
+            else:
+                raise ValueError(f"Invalid transformation option at index {i}: {option}")
+
+            #stdardize 
+            if self.stdize is not None:
+                Jacobian_std_factor =  1.0/np.std(input_Jacobian) 
```

Comment:
what is this supposed to be?  The class instance already has an attribute recording the stds as you see from the prepare_data method